### DISCUSSION TITLE
Compatibility checker between model and database

### DISCUSSION THREAD

**Discussion by kasium** [sqlalchemy/alembic#1448] [Ideas]
Background: Some time ago I made some columns in a table in model (SQLAlchemy) nullable. For reasons, no test was added to verify my change. After deploying my change, I noticed that I forgot to write an alembic migration changing the columns to nullable

Idea: Some kind of function in alembic which checks if any migration was forgot to be written. It's similar to autogeneration but with more focus on checking rather than generation. The check could be as easy as generating an autogenerated migration and verifying that it's empty

Configuration: Sometimes differences between the database and the model are by purpose. E.g. it's okay to tell SQLAlchemy something is a primary key even if it's not (the application verifies this, but not the db due to performance reasons). So there needs to be a kind of allow list

This idea would help me a lot, but I see if it's not in the scope of alembic and/or to complicated

**Comment by CaselIT** [comment]
Hi,

Have you looked into `alembic check`? That's exactly this:
> Some kind of function in alembic which checks if any migration was forgot to be written. It's similar to autogeneration but with more focus on checking rather than generation. The check could be as easy as generating an autogenerated migration and verifying that it's empty

**Reply by kasium** [reply]
Nice and thanks for the fast reply! is there any way to provide a allow-list or a kind of baseline?

**Reply by CaselIT** [reply]
you can use the functions that are available to autogenerate to ignore things: include_name and include_object
